### PR TITLE
feat: add swipe to bottomsheet and fix swipe for drawer

### PR DESCRIPTION
**Pull request by jesperwiner**

### CODE REVIEW

File: src/Molecules/BottomSheet/BottomSheet.styles.ts [hunk 1]
```diff
@@ -88,8 +88,32 @@ export const StyledBottomSheet = styled(motion.div)<{
...
+export const DragHandle = styled.div`
+  display: flex;
+  justify-content: center;
+  touch-action: manipulation;
+`;
+
+export const BottomDragArea = styled.div`
+  margin-left: ${(p) => p.theme.spacing.unit(-5)}px;
+  padding-right: ${(p) => p.theme.spacing.unit(10)}px;
+  background: ${({ theme }) => theme.colorTokens.neutral.background_default};
```
Comment (mikaelmalmqvistnordnet) [inline src/Molecules/BottomSheet/BottomSheet.styles.ts:116]: for consistency:
```suggestion
  margin-left: ${({ theme }) => theme.spacing.unit(-5)}px;
  padding-right: ${({ theme }) => theme.spacing.unit(10)}px;
  background: ${({ theme }) => theme.colorTokens.neutral.background_default};
```

File: src/Molecules/BottomSheet/BottomSheet.styles.ts [hunk 2]
```diff
@@ -88,8 +88,32 @@ export const StyledBottomSheet = styled(motion.div)<{
...
+  bottom: -100%;
   width: 100%;
   z-index: ${(p) => p.theme.zIndex.overlayInModal};
 `;
+
+export const Handle = styled.div`
+  background-color: ${({ theme }) => theme.colorTokens.neutral.icon_disabled};
+  border-radius: ${({ theme }) => theme.borderRadius20};
+  height: 4px;
+  margin-bottom: ${(p) => p.theme.spacing.unit(1)}px;
```
Comment (mikaelmalmqvistnordnet) [inline src/Molecules/BottomSheet/BottomSheet.styles.ts:103]: for consistency:

```suggestion
  margin-bottom: ${({ theme }) => theme.spacing.unit(1)}px;
```

File: src/Molecules/BottomSheet/BottomSheet.tsx [hunk 3]
```diff
@@ -19,30 +26,57 @@ const BottomSheet: React.FC<Props> = ({
...
     if (typeof onCloseExternal === 'function') {
       onCloseExternal();
     }
-  };
+  }, [onCloseExternal]);
+
+  const dragControls = useDragControls();
+
+  const startDrag = useCallback(
+    (event: any) => {
```
Comment (mikaelmalmqvistnordnet) [inline src/Molecules/BottomSheet/BottomSheet.tsx:49]:
```suggestion
    (event: TouchEvent) => {
```

File: src/Molecules/BottomSheet/BottomSheet.tsx [hunk 4]
```diff
@@ -19,30 +26,57 @@ const BottomSheet: React.FC<Props> = ({
...
+    },
+    [dragControls],
+  );
+
+  const handleClose = useCallback(() => {
+    setIsOpenInternal(false);
+
+    if (onClose) {
+      onClose();
+    }
```
Comment (mikaelmalmqvistnordnet) [inline src/Molecules/BottomSheet/BottomSheet.tsx:62]: `onClose` is always defined, so no need for this if-statement. Or did you mean to only run `onCloseExternal` here?

File: src/Molecules/Drawer/Drawer.styles.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  width: 100%;
+  height: 100%;
+  background: ${({ theme }) => theme.color.card};
+  position: fixed;
+  z-index: ${({ theme }) => theme.zIndex.overlay};
+  right: 0;
+  top: 0;
+  display: flex;
+  flex-direction: column;
+  box-shadow: 0 -10px 16px 0 ${({ theme }) => theme.color.shadowModal};
```
[the remark this answers is not in the document]
Comment (jesperwiner) [reply src/Molecules/Drawer/Drawer.styles.ts:22]: maybe, this was just moved though

File: src/Molecules/BottomSheet/BottomSheet.tsx (hunk 4 above)
Comment (jesperwiner) [reply src/Molecules/BottomSheet/BottomSheet.tsx:62]: removed it to only be a handleClose

File: src/Molecules/BottomSheet/BottomSheet.tsx [hunk 6]
```diff
@@ -62,38 +96,51 @@ const BottomSheet: React.FC<Props> = ({
...
+                      )}
+                      <Flexbox container justifyContent="space-between" alignItems="center">
+                        <Flexbox item>{title}</Flexbox>
+                        <Flexbox item>
+                          <StyledIconButton
+                            $invertedColors={invertedColors}
+                            onClick={onClose}
+                            size="s"
+                            variant="primary"
+                          >
```
[the remark this answers is not in the document]
Comment (jesperwiner) [reply src/Molecules/BottomSheet/BottomSheet.tsx]: normally we don't want untranslated aria-labels but this is also just a mobile component, not intended for desktop, thats why there is no tooltip label or something

**Review by mikaelmalmqvistnordnet — APPROVED** [review]

**Pull request merged**